nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,347 @@
+          </nuxeo-card>
+
+          <!-- LegalHold by Document Type -->
+          <nuxeo-repository-data
+            start-date="[[_formatDate(startDate)]]"
+            end-date="[[_extendEndDate(endDate)]]"
+            date-field="dc:created"
+            grouped-by="ecm:primaryType"
+            where='{"ecm:hasLegalHold": "true"}'
+            group-limit="20"
+            data="{{legalHoldByDocType}}"
+            index="[[index]]"
+          >
+          </nuxeo-repository-data>
+
+          <nuxeo-card heading="[[i18n('retentionAnalytics.legalHoldByDocType')]]">
+            <chart-pie
+              values="[[_values(legalHoldByDocType)]]"
+              labels="[[_labels(legalHoldByDocType)]]"
+              options='{ "legend": { "display": true, "position": "bottom", "labels": { "boxWidth": 12 } }, "animation": false }'
+            >
+            </chart-pie>
+          </nuxeo-card>
+        </div>
+
+        <!-- Expiration next 12 months -->
+        <nuxeo-repository-data
+          start-date="[[_formatDateNow()]]"
+          end-date="[[_formatDateFutureInMonths(12)]]"
+          with-date-intervals="month"
+          date-field="ecm:retainUntil"
+          data="{{expirationPerMonth}}"
+          index="[[index]]"
+        >
+        </nuxeo-repository-data>
+
+        <nuxeo-card class="chartBar" heading="[[i18n('retentionAnalytics.expirationNext12Months.heading')]]">
+          <chart-bar
+            labels="[[_futureExpirationLabels(expirationPerMonth)]]"
+            values="[[_values(expirationPerMonth)]]"
+            options='{ "legend": { "display": false }, "animation": false }'
+          >
+          </chart-bar>
+        </nuxeo-card>
+      </nuxeo-page>
+    `;
+  }
+
+  static get is() {
+    return 'nuxeo-retention-analytics';
+  }
+
+  static get properties() {
+    return {
+      index: {
+        type: String,
+        value: '_all',
+      },
+      startDate: {
+        type: String,
+        value: moment().subtract(3, 'year').format('YYYY-MM-DD'),
+      },
+      endDate: {
+        type: String,
+        value: moment().format('YYYY-MM-DD'),
+      },
+    };
+  }
+
+  _values(data) {
+    return [
+      data.map((entry) => {
+        if (Array.isArray(entry.value)) {
+          return entry.value.map(this._values);
+        }
+        return entry.value;
+      }),
+    ];
+  }
+
+  _labels(data) {
+    return data.map((entry) => {
+      if (Array.isArray(entry.value)) {
+        return entry.value.map(this._labels);
+      }
+      return entry.key;
+    });
+  }
+
+  _valuesRetentionStatus(noRetention, futureRetention, expiredRetention) {
+    const interminate = futureRetention.filter((entry) => entry.key === '9999-01-01T00:00:00.000Z').length;
+    return [[noRetention, futureRetention.length, expiredRetention.length, interminate]];
+  }
+
+  _labelsRetentionStatus() {
+    return ['no retention', 'active retention', 'expired retention', 'interminate retention'];
+  }
```

Comment:
In the jira ticket the labels are in upper case. Also consider to put them as a constant and use them directly in the data elements.

Reply:
What about my comment ⬆️ https://github.com/nuxeo/nuxeo-retention/pull/37#r556576396